app/controllers/admin/api/backend_apis_controller.rb

```diff
@@ -0,0 +1,146 @@
+  ##
+  ##~ op.parameters.add @parameter_access_token
+  ##~ op.parameters.add @parameter_page
+  ##~ op.parameters.add @parameter_per_page
+  ##
+  #
+  def index
+    respond_with(current_account.backend_apis.paginate(pagination_params))
+  end
+
+  ##~ e = sapi.apis.add
+  ##~ e.path = "/admin/api/backend_apis.json"
+  ##~ e.responseClass = "backend_api"
+  #
+  ##~ op = e.operations.add
+  ##~ op.httpMethod = "POST"
+  ##~ op.summary    = "Backend API Create"
+  ##~ op.description = "Creates a Backend API."
+  ##~ op.group = "backend_api"
+  #
+  ##~ op.parameters.add @parameter_access_token
+  ##~ op.parameters.add @parameter_backend_api_name
+  ##~ op.parameters.add @parameter_backend_api_description
+  ##~ op.parameters.add @parameter_backend_api_private_endpoint
+  #
+  def create
+    backend_api = current_account.backend_apis.create(backend_api_params)
+    respond_with(backend_api)
+  end
+
+  ##~ e = sapi.apis.add
+  ##~ e.path = "/admin/api/backend_apis/{id}.json"
+  ##~ e.responseClass = "backend_api"
+  #
+  ##~ op = e.operations.add
+  ##~ op.httpMethod = "GET"
+  ##~ op.summary = "Backend API Read"
+  ##~ op.description = "Returns a backend API."
+  ##~ op.group = "backend_api"
+  #
+  ##~ op.parameters.add @parameter_access_token
+  ##~ op.parameters.add @parameter_backend_api_id_by_id
+  #
+  def show
+    respond_with(backend_api)
+  end
+
+  ##~ e = sapi.apis.add
+  ##~ e.path = "/admin/api/backend_apis/{id}.json"
+  ##~ e.responseClass = "backend_api"
+  #
+  ##~ op            = e.operations.add
+  ##~ op.httpMethod = "PUT"
+  ##~ op.summary    = "Backend API Update"
+  ##~ op.description = "Updates a backend API."
+  ##~ op.group = "backend_api"
+  #
+  ##~ op.parameters.add @parameter_access_token
+  ##~ op.parameters.add @parameter_backend_api_id_by_id
+  ##~ op.parameters.add @parameter_backend_api_name
+  ##~ op.parameters.add @parameter_backend_api_description
+  ##~ op.parameters.add @parameter_backend_api_private_endpoint
+  #
+  def update
+    backend_api.update(backend_api_params)
+    respond_with(backend_api)
+  end
+
+  ##~ e = sapi.apis.add
+  ##~ e.path = "/admin/api/backend_apis/{id}.json"
+  ##~ e.responseClass = "backend_api"
+  #
+  ##~ op            = e.operations.add
+  ##~ op.httpMethod = "DELETE"
+  ##~ op.summary    = "Backend API Delete"
+  ##~ op.description = "Deletes a backend API."
+  ##~ op.group = "backend_api"
+  #
+  ##~ op.parameters.add @parameter_access_token
+  ##~ op.parameters.add @parameter_backend_api_id_by_id
+  #
+  def destroy
+    backend_api.destroy
+    respond_with(backend_api)
+  end
+
+  private
+
+  attr_reader :backend_api
+
+  def authorize
+    provider_can_use!(:api_as_product)
+    # TODO: I guess it should authorize also that it has permission for the specific action :)
+  end
+
+  def find_backend_api
```

Comment:
There is no need to find it before action
Rename it to `def backend_api` and memoize the variable